internal/brokers/broker.go

```diff
@@ -288,9 +293,9 @@ func (b Broker) parseSessionID(sessionID string) string {
 	return strings.TrimPrefix(sessionID, fmt.Sprintf("%s-", b.ID))
 }
 
-// validateAndMarshalUser checks if the specified rawMsg is a valid userinfo and marshals it as a JSON string.
-func (b Broker) validateAndMarshalUser(rawMsg json.RawMessage) (data string, err error) {
-	defer decorate.OnError(&err, "invalid user information")
+// validateUserInfoAndGenerateIDs checks if the specified rawMsg is a valid userinfo and generates the UID and GIDs.
+func validateUserInfoAndGenerateIDs(brokerName string, rawMsg json.RawMessage) (user cache.UserInfo, err error) {
+	defer decorate.OnError(&err, "invalid user information: %s", rawMsg)
```

Comment:
hum, this will read weird with an error (see golden files). What about `invalid user information provided by the broker (%s)`, then the error will be appended.